tutorials/cyber/network_mapping/lanl_network_mapping_using_rapids.ipynb

```diff
@@ -0,0 +1,801 @@
+   "source": [
+    "ds_gdfs = {\n",
+    "           'lanl_auth': auth_gdf, \n",
+    "           'lanl_flow': flow_gdf\n",
+    "          }"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Build edges dataframe\n",
+    "\n",
+    "In the cell below, reference each datasource and its corresponding edge configuration to build a new dataframe containing edges. This dataframe will notably contain `srcCol` and `dstCol` along with other reference data."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "edges_gdf = None\n",
+    "\n",
+    "for ds in edge_defs:\n",
+    "    \n",
+    "    ds_gdf = ds_gdfs[ds['dataSource']]\n",
+    "    \n",
+    "    for e in ds[\"edges\"]:\n",
+    "        \n",
+    "        evtCols = ds[\"stringCols\"].copy()\n",
+    "        \n",
+    "        evtCols.append(e[\"srcCol\"])\n",
+    "        evtCols.append(e[\"dstCol\"])\n",
+    "        evtCols.append(ds[\"timeCol\"])\n",
+    "        evtCols = list(set(evtCols))\n",
+    "        \n",
+    "        eventsDF = ds_gdf\n",
+    "        eventsDF = eventsDF[evtCols]\n",
+    "        \n",
+    "        # Apply filters indicated in the edge configuration file\n",
+    "        if 'filters' in e:\n",
+    "            for f in e['filters']:\n",
+    "                eventsDF = eventsDF[eventsDF[f['key']].str.contains(f['value']) == True]\n",
+    "        \n",
+    "        # Remove any None values\n",
+    "        src_idx = eventsDF[e['srcCol']].str.contains(\"None\")\n",
+    "        if len(eventsDF[src_idx])>0:\n",
+    "            eventsDF = eventsDF[src_idx==False]\n",
+    "        \n",
+    "        dst_idx = eventsDF[e['dstCol']].str.contains(\"None\")\n",
+    "        if len(eventsDF[dst_idx])>0:\n",
+    "            eventsDF = eventsDF[dst_idx==False]        \n",
+    "                \n",
+    "        evt_edges_gdf = cudf.dataframe.DataFrame()\n",
+    "        evt_edges_gdf['src'] = eventsDF[e[\"srcCol\"]]\n",
+    "        evt_edges_gdf['dst'] = eventsDF[e[\"dstCol\"]]\n",
+    "        \n",
+    "        # Adjust time to recent date (LANL data source begins at 1 second)\n",
+    "        evt_edges_gdf['time'] = eventsDF[ds[\"timeCol\"]]+1442131200\n",
+    "        \n",
+    "        evt_edges_gdf['src_node_type'] = e[\"srcNodeType\"]\n",
+    "        evt_edges_gdf['dst_node_type'] = e[\"dstNodeType\"]\n",
+    "        evt_edges_gdf['relationship'] = e[\"relationship\"]\n",
+    "        evt_edges_gdf['data_source'] = ds[\"dataSource\"]\n",
+    "        \n",
+    "        if edges_gdf is None:\n",
+    "            edges_gdf = evt_edges_gdf\n",
+    "        else:\n",
+    "            edges_gdf = cudf.concat([edges_gdf, evt_edges_gdf])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Use pandas to drop duplicates as this is not yet available in cudf for strings"
```

Comment:
Please make a feature request on cudf for  `drop_duplicates`! Feature requests like these are exactly what we want to come out of these workflows.

Reply:
Thanks! https://github.com/rapidsai/cudf/issues/1467